TACC/Core-CMS-Custom
# task/WP-293:  Renewal Registration Form

## Overview
Added ability to pre-load data into /register/request-to-submit form for annual renewal purposes
…

## Related


- [WP-293](https://jira.tacc.utexas.edu/browse/WP-293)
- [WP-284](https://jira.tacc.utexas.edu/browse/WP-284)
…

## Changes
- Modified `get()` method for registration form to check for registration id in request
- If `reg_id` in request, grabs corresponding registration data from database and loads to template
- Database functions for writing new registration entities and contacts modified to handle loading "old" registration data but writing new record
- For multiple places that registration data is loaded to templates, moved data formatting out from admin registration table view to its own util script
…

## Testing
** Will require some hard coding, not yet connected to listing page action which will send registration id **
1. In `src/apps/registrations/views.py`, replace lines 24 and 25 with the following:
    <details><summary>Snippet</summary>

        if True:
            reg_id = 72 # record 'Renewal Test 6'

    </details>
2. Go to https://localhost:8000/register/request-to-submit/
3. Confirm that the data pre-loaded to the form matches the data for the record `Renewal Test 6` on https://localhost:8000/administration/list-registration-requests AND that the fields for any entities' 'Coverage Estimates' are not pre-loaded with data
4. Make changes and fill any required fields and confirm you can submit form successfully
5. After form submission, go to https://localhost:8000/administration/list-registration-requests and confirm your new record is on the table and, with 'View Record' in the drop down, that the full record you submitted is shown
6. Now, revert your changes from step 1 and go back to https://localhost:8000/register/request-to-submit/ and confirm the form is shown will all blank fields


## UI

…
## Notes
- Follow on task required to provide 'Renew Registration' action on listing page from [WP-284](https://github.com/TACC/Core-CMS-Custom/pull/232) which will direct to this pre-loaded form
